### ISSUE TITLE
Parsing issues with @image directive. Fatal Error

### ISSUE THREAD

**Issue by christianmagill** [Log1x/sage-directives#42]
I've run into parsing issues with the @image directive.

Case 1:

When using the directive as follows:

`@image($image_id, 'medium', $alt)`

The alt attribute is assigned the literal string "$alt" instead of the variable.

The resulting code ends up being...

```
<?php echo wp_get_attachment_image(
                $image_id,
                'medium',
                false,
                ['alt' => '$alt']
            ); ?>
```

Case 2:

When using the directive as follows:

@image($image_id, 'medium', ['alt' => $alt, class='alignright'] )

This one results in a Fatal Error

Fatal error: Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: syntax error, unexpected '','' (T_CONSTANT_ENCAPSED_STRING), expecting ']'

The resulting code is...

```
<?php echo wp_get_attachment_image(
                $image_id,
                'medium',
                false,
                ['alt' => $alt','class' => 'alignright']
            ); ?>

```

**Comment by ajcsilva** [comment]
Any updates or workarounds for this?

**Comment by 1pix** [comment]
Hi there,
Any update or possible workaround ?
Tried a lot of combination, still no solution on my side :/
Thanks

probably related to #30

**Comment by 1pix** [comment]
After trying many things in `Log1x\SageDirectives\Util` , and saw your big comment in method "isArray" :)

Maybe some ideas here, a bit old, not tested, but putting it here as refs: 
- **"Using json format to passe data to the directive"**
@see "Another solution" https://stackoverflow.com/a/53218523

No time at the moment but will try if I can.

Cheers and thanks you @Log1x for the huge and so usefull work you do, as ever.

**Comment by thomasjonas** [comment]
Is this issue really solved? I've tried running version 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4 and 2.0.5 but I still get the following error: ` Uncaught ParseError: syntax error, unexpected single-quoted string ",", expecting "]" `

This is in my blade file:
```
@php
        $alt = $item['logo']['alt'] ? $item['logo']['alt'] : 'Logo';
@endphp

@image($item['logo']['ID'], 'medium', ['alt' => $alt, 'class' => 'h-12 w-auto mix-blend-darken'])
```

And this is what it compiles to:

```
 <?php $__imageDirective = $item['logo']['ID']; ?><?php echo wp_get_attachment_image(
                            $__imageDirective,
                            'medium',
                            false,
                            ['alt' => $alt','class' => 'h-12 w-auto mix-blend-darken']
                        ); ?>
```

The following:

`@image($item['logo']['ID'], 'medium', ['alt' => 'alt text', 'class' => 'h-12 w-auto mix-blend-darken'])`

compiles to:

```
<?php $__imageDirective = $item['logo']['ID']; ?><?php echo wp_get_attachment_image(
                            $__imageDirective,
                            'medium',
                            false,
                            ['alt' => 'alt text','class' => 'h-12 w-auto mix-blend-darken']
                        ); ?>
```

**Comment by Log1x** [comment]
It was supposed to be but v2.0 was a haul so it's possible it got overlooked. I'd appreciate a PR.
